support/build/Makefile.rules

```
@@ -59,8 +59,8 @@ endef
 ################################################################################
 
 define mk_sub_build_dir =
-$(if $(shell mkdir -p "$(BUILD_DIR)/$(1)" && cd "$(BUILD_DIR)/$(1)" >/dev/null && pwd),,\
-     $(error could not create directory "$(BUILD_DIR)/$(1)"))
+$(if $(shell mkdir -p "$(BUILD_DIR)/$(notdir $(1))" && cd "$(BUILD_DIR)/$(notdir $(1))" >/dev/null && pwd),,\
```

If `addlib` and friends expect a name, would make more sense to return an error here if someone passes a path instead of silently handling with `notdir`?

---

I'd prefer avoiding this because of more costly string parsing for parameters with Make.